corehq/apps/reports/tasks.py

```diff
@@ -94,6 +94,11 @@ def send_delayed_report(report_id):
 def send_report(notification_id):
     notification = ReportNotification.get(notification_id)
 
+    # If the report's start date is later than today, return and do not send the email
+    if notification.start_date and notification.start_date > datetime.today().date():
+        daily_reports()
```

Comment:
@gbova I just stumbled across this. I don't think I follow the logic, and from the face of it I'd think this could result in a recursive loop, with some reports getting sent out many times. What's the idea behind calling `daily_reports` here?